lib/app/app.ts

```diff
@@ -0,0 +1,35 @@
+import {GenericConstructor} from "../interfaces";
+import {Request} from "../request";
+import {Dispatcher} from "./dispatcher";
+import {Response} from "../response";
+import {IContainer} from "./IContainer";
+import {IRouter} from "../routing";
+import {IApp} from "./IApp";
+
+
+export class App implements IApp {
+
+    constructor(protected container: IContainer, protected router: IRouter) {
+    }
+
+    resolve<T>(constructor: GenericConstructor<T>): T {
+        return this.container.resolve(constructor);
+    }
+
+    async handleRequest(request: Request): Promise<Response> {
+        // TODO: better response.
+        try {
+            const data = this.router.getRouteData(request);
+            await Dispatcher.dispatchMiddleware(this.container, request, data.route.middleware || []);
```

Comment:
Any reason we need to do this `|| []` here? Could probably default this in the method itself